File: server/ruby/app/infrastructure/mappers/payment_details/payment_product3208_specific_output_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PaymentProduct3208SpecificOutputMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::PaymentProduct3208SpecificOutput.new
          dto.buyer_compliant_bank_message = response.buyer_compliant_bank_message

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/payment_product_3208_specific_output_mapper.py
from typing import Optional

from app.application.domain.payments.payment_details.payment_product_3208_specific_output import PaymentProduct3208SpecificOutput
from onlinepayments.sdk.domain.payment_product3208_specific_output import PaymentProduct3208SpecificOutput as PaymentProduct3208SpecificOutputSdk

class PaymentProduct3208SpecificOutputMapper:

    @staticmethod
    def map_from_sdk_response(
        response: Optional[PaymentProduct3208SpecificOutputSdk],
    ) -> Optional[PaymentProduct3208SpecificOutput]:
        if response is None:
            return None

        dto = PaymentProduct3208SpecificOutput()
        dto.buyer_compliant_bank_message = response.buyer_compliant_bank_message

        return dto